go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -532,10 +536,339 @@ func (c *cnxn) GetInfo(ctx context.Context, infoCodes []adbc.InfoCode) (array.Re
 // All non-empty, non-nil strings should be a search pattern (as described
 // earlier).
 func (c *cnxn) GetObjects(ctx context.Context, depth adbc.ObjectDepth, catalog *string, dbSchema *string, tableName *string, columnName *string, tableType []string) (array.RecordReader, error) {
-	return nil, adbc.Error{
-		Msg:  "Not implemented: GetObjects",
-		Code: adbc.StatusNotImplemented,
+	builder := array.NewRecordBuilder(c.db.alloc, adbc.GetObjectsSchema)
+	defer builder.Release()
+
+	// To avoid an N+1 query problem, we assume result sets here will fit in memory and build up a single response.
+	info, err := c.cl.GetCatalogs(ctx)
+	if err != nil {
+		return nil, adbcFromFlightStatus(err)
+	}
+
+	rdr, err := c.readInfo(ctx, schema_ref.Catalogs, info)
```

Comment:
we're guaranteed that there'll only ever be one endpoint for this?

Reply:
readInfo uses the statement reader, so we don't need to make that assumption